whisperv6/whisper.go

```diff
@@ -800,8 +813,17 @@ func (whisper *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 			log.Warn("unxepected status message received", "peer", p.peer.ID())
 		case messagesCode:
 			// decode the contained envelopes
+			data, err := ioutil.ReadAll(packet.Payload)
+			if err != nil {
+				log.Warn("failed to read envelopes data", "peer", p.peer.ID(), "error", err)
+				return errors.New("invalid enveloopes")
+			}
+			if !whisper.disableConfirmations {
+				whisper.sendConfirmation(p.peer.ID(), rw, data)
```

Comment:
```suggestion
				go whisper.sendConfirmation(p.peer.ID(), rw, data)
```

I would actually prefer to be explicit on the caller side. It's much easier to test a method that does not contain `go` calls.